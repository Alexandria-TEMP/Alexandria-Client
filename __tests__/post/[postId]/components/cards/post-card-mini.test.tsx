// import { expect, describe, it } from "@jest/globals";
// import { render, screen, waitFor } from "@testing-library/react";
import fetchPostData from "@/lib/api/services/post-api";
// import PostCardMini from "@/post/[postId]/components/cards/post-card-mini";
import { dummyPost } from "~/__tests__/__utils__/dummys";

jest.mock("@/lib/api/services/post-api");

// Mock useRouter so it exists when component is rendered
jest.mock("next/navigation");

describe("PostCardMini", () => {
  (fetchPostData as jest.Mock).mockResolvedValue(dummyPost);

  it("PLACEHOLDER TEST, REMOVE ME", () => {});

  // TODO
  // it("shows the post title after load", async () => {
  //   render(<PostCardMini id={dummyPost.id} />);

  //   // Wait for the data to load and skeleton to be gone
  //   await waitFor(() => {
  //     expect(screen.getByRole("heading")).toBeInTheDocument();
  //   });

  //   expect(screen.getByText(dummyPost.title)).toBeInTheDocument();
  //   expect(screen.getByText(dummyPost.status)).toBeInTheDocument();
  // });
});
