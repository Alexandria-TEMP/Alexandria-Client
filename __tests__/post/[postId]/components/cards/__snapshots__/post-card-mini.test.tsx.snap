// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCardMini matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col relative overflow-hidden height-auto text-foreground box-border bg-content1 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large w-full transition-transform-background motion-reduce:transition-none"
    tabindex="-1"
  >
    <div
      class="relative flex w-full p-3 flex-auto flex-col place-content-inherit align-items-inherit h-auto break-words text-left overflow-y-auto subpixel-antialiased"
    >
      <h2>
        Post title
      </h2>
    </div>
    <div
      class="p-3 h-auto w-full items-center overflow-hidden color-inherit subpixel-antialiased rounded-b-large flex flex-row"
    >
      <div
        class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap px-1 h-7 text-small rounded-full bg-default text-default-foreground"
      >
        <span
          class="flex-1 text-inherit font-normal px-2"
        >
          Reviewed
        </span>
      </div>
      <div
        class="grow"
      />
      <button
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-medium bg-transparent px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none border-primary text-primary hover:!text-primary-foreground hover:!bg-primary"
        type="button"
      >
        Return to post
      </button>
    </div>
  </div>
</div>
`;
